tests/tests.py

```diff
@@ -696,9 +696,10 @@ class TaggableManagerDirectCustomPKTestCase(TaggableManagerTestCase):
     taggeditem_model = TaggedCustomPKFood
 
     def test_require_pk(self):
-        # TODO with a charfield pk, pk is never None, so taggit has no way to
-        # tell if the instance is saved or not
-        pass
+        food_instance = self.food_model(name='foobar')
+        with self.assertRaises(self.food_model.DoesNotExist):
+            food_instance.refresh_from_db()
+            food_instance.pk
```

Comment:
This line and the other `.pk` line are never executed as the previous line throws the exception. They can be removed.